doc/aspice_swe3_4/example/iceoryx_component/source/example_component/example_base_class.hpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef IOX_DOC_EXAMPLE_COMPONENT_EXAMPLE_BASE_CLASS_HPP
+#define IOX_DOC_EXAMPLE_COMPONENT_EXAMPLE_BASE_CLASS_HPP
+
+namespace example
+{
+/// @brief Short description
+/// @code
+///   minimalistic working example which uses all public methods
+/// @endcode
+/// @swcomponent example_component
+void someOtherFreeFunction() noexcept;
+
+/// @brief Base class which is not part of the public API
+/// @details Detailed description
```

Comment:
Do we want those details for every class? Or is it optional? Or only for "public" classes which are supposed to be used by the client (i.e. because are return types of public methods). Doing this with a lot of detail for all classes could be unrealistic.

Reply:
@details is always optional, it can be used to add more text because the brief description should be ideally a oneliner